### PR TITLE
Add lock status to CLI

### PR DESCRIPTION
**Pull request by FRSTR**
Add lock_status to CLI, also update tests

### CODE REVIEW

**Comment by FRSTR** [comment]
@epenet 
I've added a function to get lock-status next to location under "status" output for CLI, and it fail tests (as I did not update them yet).  I don't thiink I get how fixtures work in  tests/cli/test_vehicle.py::test_vehicle_status. What files should be updated so that the tests don't fail with the new line under status output in CLI?

**Comment by FRSTR** [comment]
@epenet could you please take a look why it fails to load lock-status endpoint in tests?
```
        result = cli_runner.invoke(__main__.main, "status")
>       assert result.exit_code == 0, result.exception
E       AssertionError: ClientConnectionError('Connection refused: GET https://api-wired-prod-1-euw1.wrd-aws.com/commerce/v1/accounts/account-id-1/kamereon/kca/car-adapter/v1/cars/VF1AAAAA555777999/lock-status?country=FR')
```
I reviewed couple times and to mee it seems everything should be in place now for tests to pass, but obviously something is wrong..

**Comment by epenet** [comment]
You need to add the lock status to the standard test injection:
https://github.com/hacf-fr/renault-api/blob/edc34b9424ebb87ba006c6271da45e8e73c891e2/tests/fixtures.py#L442-L448

File: src/renault_api/kamereon/models.py [hunk 1]
```diff
@@ -49,13 +49,17 @@
     "X101VE": {  # ZOE phase 1
         "reports-in-watts": True,
         "support-endpoint-location": False,
+        "support-endpoint-lock-status": False,
```
Comment (epenet) [inline src/renault_api/kamereon/models.py:52]: Can you set this on GATEWAY_SPECIFICATIONS also?

File: src/renault_api/kamereon/models.py [hunk 2]
```diff
@@ -49,13 +49,17 @@
...
     },
     "X102VE": {  # ZOE phase 2
         "warns-on-method-set_ac_stop": "Action `cancel` on endpoint `hvac-start` may not be supported on this model.",  # noqa
     },
     "XJB1SU": {  # CAPTUR II
         "support-endpoint-hvac-status": False,
     },
+    "XJD1SU": {  # NEW DUSTER
+        "support-endpoint-hvac-start": False,
+    },
```
Comment (epenet) [inline src/renault_api/kamereon/models.py:62]: I think this shouldn't be mixed in with "lock_status" updates.
It should be kept for another PR.

File: src/renault_api/kamereon/models.py (hunk 1 above)
Comment (FRSTR) [reply src/renault_api/kamereon/models.py:52]: Done

File: tests/cli/test_vehicle.py [hunk 3]
```diff
@@ -112,6 +115,7 @@
         "GPS Latitude      48.1234567\n"
         "GPS Longitude     11.1234567\n"
         "GPS last updated  2020-02-18 17:58:38\n"
+        "Lock Status       locked"
```
Comment (epenet) [inline tests/cli/test_vehicle.py:118]: Missing new line (`\n`) in the expected output:
```suggestion
        "Lock Status       locked\n"
```

File: tests/cli/test_vehicle.py [hunk 4]
```diff
@@ -69,6 +71,7 @@
...
```
Comment (epenet) [inline tests/cli/test_vehicle.py:74]: Missing new line (`\n`) in the expected output:
```suggestion
        "Lock Status       locked\n"
```

File: tests/cli/test_vehicle.py [hunk 5]
```diff
@@ -55,6 +56,7 @@
...
```
Comment (epenet) [inline tests/cli/test_vehicle.py:59]: Missing new line (`\n`) in the expected output:
```suggestion
        "Lock Status       locked\n"
```

File: tests/cli/test_vehicle.py [hunk 6]
```diff
@@ -39,6 +39,7 @@
...
```
Comment (epenet) [inline tests/cli/test_vehicle.py:42]: Missing new line (`\n`) in the expected output:
```suggestion
        "Lock Status       locked\n"
```

**Comment by epenet** [comment]
LGTM.
Do you think it is useful to add the door status, and the last updated timestamp?

**Comment by FRSTR** [comment]
Well, the timestamp is useful. The door statuses - not sure, especially if they don't work. 

Maybe as a separate feature and later? I'm a bit exhausted as of now, I spend hours to setup python env for this project and did not get too far. Currently just coding via notepad basically.

**Comment by epenet** [comment]
I'm not sure what OS you are using, but I find that the easiest editor for this is VSCode (especially over WSL).

**Comment by FRSTR** [comment]
This looks nice, I happen to use WSL and tried setting up inside. This way looks to be more friendly, thank you.

**Comment by epenet** [comment]
In VSCode Preferences>Settings you can enable "Editor: Format On Save" and it should fix all the formatting issues automatically using `black`

**Comment by FRSTR** [comment]
Ok. I'm figuring out how it works, and I was finalyy able to configure python packages.

I think the next thing I'll try is to write some dirty prototype with SRP to make engine start. I doubt it will not be a mess, but if works - I guess we can figure out how to make it pretty after that.

**Comment by epenet** [comment]
Can you please add the timestamp for the lock status when you get a chance?
Now that #502 has been merged, I'm just waiting on this one to publish the next release.

**Comment by FRSTR** [comment]
Ok, I'll try to add it to CLI today

File: src/renault_api/cli/renault_vehicle.py [hunk 7]
```diff
@@ -292,6 +293,27 @@ async def update_location(
...
+        if not await vehicle.supports_endpoint("lock-status"):
+            return
+        response = await vehicle.get_lock_status()
+    except QuotaLimitException as exc:  # pragma: no cover
+        raise click.ClickException(repr(exc)) from exc
+    except KamereonResponseException as exc:  # pragma: no cover
+        click.echo(f"lock status: {exc.error_details}", err=True)
+    else:
+        items = [
+            ("Lock Status", response.lockStatus, None),
```
Comment (epenet) [inline src/renault_api/cli/renault_vehicle.py:310]:
```suggestion
            ("Lock Status", response.lockStatus, None),
            ("Lock last updated", response.lastUpdateTime, "tzdatetime"),
```

**Comment by FRSTR** [comment]
@epenet could you take a look into these failures with status output?
I guess some spaces or dashes should be added, but where and how not to break it even further?

```
>           assert EXPECTED_STATUS[filename] == result.output
E           AssertionError: assert '------------...-----------\n' == '------------...-----------\n'
E             - -----------------  -------------------------
E             ?                 -
E             + ----------------  -------------------------
E             - Battery level      50 %
E             ?                   -
E             + Battery level     50 %
E             - Last updated       2020-11-17 09:06:48
E             ?             -
E             + Last updated      2020-11-17 09:06:48
E             - Range estimate     128 km
E             ?                   -
E             + Range estimate    128 km
E             - Plug state         PlugState.UNPLUGGED
E             ?           -
E             + Plug state        PlugState.UNPLUGGED
E             - Charging state     ChargeState.NOT_IN_CHARGE
E             ?               -
E             + Charging state    ChargeState.NOT_IN_CHARGE
E             - Charge mode        always
E             ?            -
E             + Charge mode       always
E             - Total mileage      5566.78 km
E             ?                   -
E             + Total mileage     5566.78 km
E             - Fuel autonomy      35.0 km
E             ?                   -
E             + Fuel autonomy     35.0 km
E             - Fuel quantity      3.0 L
E             ?                   -
E             + Fuel quantity     3.0 L
E             - GPS Latitude       48.1[234](https://github.com/hacf-fr/renault-api/runs/5066738045?check_suite_focus=true#step:9:234)567
E             ?             -
E             + GPS Latitude      48.1234567
E             - GPS Longitude      11.1234567
E             ?              -
E             + GPS Longitude     11.1234567
E             - GPS last updated   2020-02-18 17:58:38
E             ?                 -
E             + GPS last updated  2020-02-18 17:58:38
E             - Lock status        locked
E             ?            -
E             + Lock status       locked
E             - Lock last updated  2022-02-02 14:51:13
E             ?                  -             ^
E             + Lock last updated 2022-02-02 13:51:13
E             ?                               ^
E             - -----------------  -------------------------
E             ?                 -
E             + ----------------  -------------------------
```

File: tests/cli/test_vehicle.py [hunk 8]
```diff
@@ -112,6 +118,8 @@
         "GPS Latitude      48.1234567\n"
         "GPS Longitude     11.1234567\n"
         "GPS last updated  2020-02-18 17:58:38\n"
+        "Lock status       locked\n"
+        "Lock last updated 2022-02-02 13:51:13\n"
```
Comment (epenet) [inline tests/cli/test_vehicle.py:122]: You need to add just one space, but on all the lines.
```suggestion
        "Lock status        locked\n"
        "Lock last updated  2022-02-02 13:51:13\n"
```

You will also need to add one dash on the left part of all the corresponding header and footer lines.

Comment (FRSTR) [reply tests/cli/test_vehicle.py]: 👍

**Comment by epenet** [comment]
Note: if you have `poetry` installed you can easily run the tests locally :

- Install the renault-api CLI: `poetry install -E cli`
- Run the tests: `poetry run pytest --cov`

There is also a shortcut to start the tests inside VSCode by using the following menu:
`Terminal` > `Run Task` > `Run tests with coverage`

If poetry is not installed yet, you can use this to install
`curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python3 -`

**Comment by epenet** [comment]
The last error seems to be a timezone issue...
The tests are forced to run in CET (UTC+1) :)

**Comment by FRSTR** [comment]
I was able to run nox with tests recently, but they seems to provide slightly different output, So I'm also checking here.

And to fix the timezone I simply add 1 hour to the fixture I guess. I'll try

**Review by epenet — APPROVED** [review]
Thanks @FRSTR 👍
